src/geometry.cpp

```diff
@@ -79,4 +82,102 @@ namespace geometry {
 	  return v;
 
 	}
+
+	std::vector<std::vector<int>> bresenham_conversion(const Cell &start, const Cell &end){
+
+	  // Vector of points
+	  std::vector<std::vector<int>> v;
+
+	  // Beginning point
+	  int p0 [2] = {start.x, start.y};
+
+	  // Ending point
+	  int p1 [2] = {end.x, end.y};
+
+	  // Change in x and y from beginning to end
+	  int deltas [2] = {(end.x - start.x), (end.y - start.y)};
+
+	  // Absolute values of  the changes
+	  int changes [2] = {abs(deltas[0]),abs(deltas[1])};
+
+	  // Lambda that emulates signum
+	  auto sgn = [](auto a){return (a > 0) - (a < 0);};
+
+	  // Which direction to step towards for the x and y functions
+	  int steps [2] = {sgn(deltas[0]), sgn(deltas[1])};
+	  
+	  // Array that dictates the dominant direction
+	  int idx [2] = {0};
+
+	  if (changes[0] > changes[1]) {
+	    idx[0] = 0;
+	    idx[1] = 1;
+	  } else {
+	    idx[0] = 1;
+	    idx[1] = 0;
+	  }
+
+	  // error scaled off of change in dominant direction
+	  float err = (float)changes[idx[0]]/2.0;
+
+	  // points that will be pushed back
+	  int point [2] = {p0[0], p0[1]};
+
+	  // while the dominant direction coordinate isn't equal to the ending coordinate
+	  while (point[idx[0]] != p1[idx[0]]) {
+
+	    // Create and push back vector into final vector
+	    std::vector<int> temp {point[0], point[1]};
+
+	    v.push_back(temp);
+
+	    // bresenham magic
+	    err -= changes[idx[1]];
+
+	    if (err < 0) {
+
+	      point[idx[1]] += steps[idx[1]];
+
+	      err += changes[idx[0]];
+
+	    }
+
+	    point[idx[0]] += steps[idx[0]];
+
+	  }
+
+	  // Push final point
+	  std::vector<int> temp {p1[0], p1[1]};
+
+	  v.push_back(temp);
+
+	  return v;
+
+	}
+
+	std::vector<Cell> raytraceline_conversion(const Cell& start, const Cell& end, size_t max_ray_length) {
+	  // Get the interpolated pixels bresenham
+	  const auto pixels_as_ints = bresenham_conversion(start.x, start.y, end.x, end.y);
+
+	  // Convert std::vector<std::vector<int>> to std::vector<Cell>
+	  std::vector<Cell> pixels_as_cells;
+
+	  // gonna cut off the vector at max length, I don't have enough brainpower to move
+	  // back the endpoint currently
+	  for (const auto &pixel:pixels_as_ints) {
+
+		if (pixels_as_cells.size() >= max_ray_length) {
+			return pixels_as_cells;
+		}
+
+	    Cell temp = {pixel[0], pixel[1]};
+	    pixels_as_cells.emplace_back(temp);
+
+	  }
+	  
+	  return pixels_as_cells;
```

Comment:
If you use the `Cell` based version of `bresenham` then this could just be
`return {pixels_as_ints.begin(), pixels_as_ints.begin() + max_ray_length}`
Though you'll also have to do something like
`const auto end_index = std::max(pixels_as_ints.size(), max_ray_length)`
Though, I don't really like that name. All this to say you won't need the `for` loop

Reply:
Done